src/Datadog.Trace.Core/TraceContext.cs

```diff
@@ -1,26 +1,29 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
-using Datadog.Trace.Logging;
-using Datadog.Trace.PlatformHelpers;
 
 namespace Datadog.Trace
 {
     internal class TraceContext : ITraceContext
     {
-        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<TraceContext>();
+        private static readonly ICoreLogger Log = CoreLogging.For<TraceContext>();
+        private static Action<Span> _decorateRootSpan = null;
 
         private readonly DateTimeOffset _utcStart = DateTimeOffset.UtcNow;
         private readonly Stopwatch _stopwatch = Stopwatch.StartNew();
         private readonly List<Span> _spans = new List<Span>();
 
+        private readonly Action<Span[]> _submit = null;
+        private readonly Func<Span, SamplingPriority> _getSamplingPriority;
+
         private int _openSpans;
         private SamplingPriority? _samplingPriority;
         private bool _samplingPriorityLocked;
 
-        public TraceContext(IDatadogTracer tracer)
+        public TraceContext(Action<Span[]> submit, Func<Span, SamplingPriority> getSamplingPriority)
```

Comment:
Why are we removing the `IDatadogTracer` parameter here in favor of delegates?

Reply:
I did this in favor of moving `IDatadogTracer` to `Datadog.Trace.Core` as it meant fewer code changes and fewer reasons to change in the future.

Happy to accomplish this however, but this felt like a reasonable solution to me, as the minimum external requirements are: (1) How to submit (2) How to get sampling rate

Reply:
Perhaps using composition on `IDatadogTracer` by having a core interface that implements those two requirements would be cleaner....
`ICoreDatadogTracer`

Reply:
I think another reason I originally did this is because these methods came from two separate interfaces.

Reply:
Yeah, `IDatadogTracer` lives in core, but not the sampler.

Reply:
I added a new strategy interface.